src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
     };
 }
 
+#[derive(Serialize, Deserialize)]
+struct LurkData<F: LurkField> {
+    z_ptr: ZPtr<F>,
+    z_dag: ZDag<F>,
+}
+
+impl<F: LurkField> HasFieldModulus for LurkData<F> {
+    fn field_modulus() -> String {
+        F::MODULUS.to_string()
+    }
+}
+
+/// Returns a `Utf8PathBuf` from a pointer
+///
+/// # Errors
+/// Errors if a string can't be fetched with the pointer
+fn get_path<F: LurkField>(repl: &Repl<F>, path: &Ptr) -> Result<Utf8PathBuf> {
+    let Some(path) = repl.store.fetch_string(path) else {
+        bail!(
+            "Path must be a string. Got {}",
+            path.fmt_to_string(&repl.store, &repl.state.borrow())
+        )
+    };
+    Ok(Utf8PathBuf::from(path))
+}
+
+impl MetaCmd<F> {
+    const DUMP_DATA: MetaCmd<F> = MetaCmd {
+        name: "dump-data",
+        summary: "Write Lurk data to the file system",
+        format: "!(dump-data <expr> <string>)",
+        description: &[],
+        example: &["!(dump-data (+ 1 1) \"my_file\")"],
+        run: |repl, args| {
+            let (expr, path) = repl.peek2(args)?;
+            let path = get_path(repl, &path)?;
+            let (io, ..) = repl
+                .eval_expr(expr)
+                .with_context(|| "evaluating predicate")?;
+            let mut z_dag = ZDag::default();
+            let z_ptr = z_dag.populate_with(&io[0], &repl.store, &mut Default::default());
+            dump(LurkData { z_ptr, z_dag }, &path)
+        },
+    };
+}
+
+impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+    const LOAD_DATA: MetaCmd<F> = MetaCmd {
+        name: "load-data",
+        summary: "Read Lurk data from the file system and bind it to a symbol",
```

Comment:
I don't think a command called `load-data` should also create a binding. Ideally, this would just return the data. I understand that we can't currently intermix Lurk and meta-Lurk in the natural way that would make this useful. If you want to make a hybrid command that loads data *and* creates a top-level lexical binding — equivalent to `!(def var !(load-data file))` it should have an appropriate name.

Reply:
Please suggest a name

Reply:
`load-and-bind-data`?

Reply:
`!(def-loaded …)` maybe?

Reply:
I guess this exposes that you're somewhat overloading `load` (since the usual meaning implies evaluation), so maybe it needs to be `!(def-loaded-data …)`, with `!(def-loaded …)` reserved for the case of actually loading.

You could probably use `(def-load-data …)` instead of adding the suffix if that makes more sense to you.

Reply:
`def-loaded` is good

Btw, why the dash between "def" and "load"

Reply:
I wrote it that way because when I wrote `defloaded`, my brain pronounced it `de-floaded`. And fully-hyphenating would be the right way anyway if using the longer form(s).

Also, I don't thinkg `def-loaded` *is* good — as I described above — since this is exactly the form we would want to use for what would otherwise (and should eventually be) expressible as `!(def name (load file))`.

Reply:
Yeah, crossed messages